vignettes/readr.Rmd

```diff
@@ -160,61 +159,46 @@ readr comes with five parsers for rectangular file formats:
 * `read_fwf()` for fixed-width files
 * `read_log()` for web log files
 
-Each of these functions firsts calls `spec_xxx()` (as described above), and then parses the file according to that column specification:
+If readr finds problems with parsing your data, it will emit a warning message.
+For example, in our `challenge.csv` file, the column type we expected from our data was a column of dates and a column of doubles.
 
 ```{r}
 df1 <- read_csv(readr_example("challenge.csv"))
 ```
 
-The rectangular parsing functions almost always succeed; they'll only fail if the format is severely messed up. Instead, readr will generate a data frame of problems. The first few will be printed out, and you can access them all with `problems()`:
+The data was successfully imported, but we get a warning message about parsing issues.
+To view more details about these parsing issues, call `problems()` on your data frame.
+This will generate a data frame with information about where the problem was found, and why.
```

Comment:
```suggestion
This will generate a data frame with information about the parsing problems, with one row per problem.
```

The previous proposed wording seemed too specific to this example, i.e. that there would be exactly 1 problem.